File: flutter-boilerplate/lib/validators/forms/invite.dart
String? validateEmail(String? value) {
  if (value == null || value.isEmpty) return null;
  final emailRegex = RegExp(r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$');
  if (!emailRegex.hasMatch(value)) return 'Invalid email';
  return null;
}

String? validateRole(String? value) {
  if (value == null || value.isEmpty) return 'Select a role';
  if (!['member', 'admin', 'owner'].contains(value)) return 'Invalid role';
  return null;
}

String? validateMessage(String? value) {
  if (value == null || value.isEmpty) return null;
  if (value.length > 1000) return 'Message must be at most 1000 characters';
  return null;
}

File: tanstack-boilerplate/src/validators/forms/invite.ts
import { z } from "zod";

const DEFAULT_INVITE_T: Record<string, string> = {
  emailRequired: "At least one email required",
  emailInvalid: "Invalid email",
  roleRequired: "Select a role",
};

export const inviteSchema = createInviteSchema(DEFAULT_INVITE_T);

export function createInviteSchema(t: Record<string, string>) {
  return z.object({
    emails: z
      .array(z.string().email(t.emailInvalid ?? "Invalid email"))
      .min(1, t.emailRequired ?? "At least one email required"),
    role: z.enum(["member", "admin", "owner"], {
      message: t.roleRequired ?? "Select a role",
    }),
    message: z.string().max(1000).optional(),
  });
}
